File: RailgunVB/Core/Api/RandomCat.vb
Imports System.IO
Imports System.Net.Http
Imports RailgunVB.Core.Configuration

Namespace Core.Api
    
    Public Class RandomCat
    
        Private ReadOnly _client As New HttpClient
        Private Const BaseUrl As String = "http://thecatapi.com/api/images/"
        Private ReadOnly _apiKey As String
        
        Public Sub New(config As MasterConfig)
            _apiKey = config.RandomCatApiToken
        End Sub
        
        Public Async Function GetRandomCatAsync() As Task(Of Stream)
            Return Await _client.GetStreamAsync($"{BaseUrl}get?api_key={_apiKey}&type=png&size=med")
        End Function
        
    End Class
    
End NameSpace

File: Railgun/Apis/RandomCat/RandomCat.cs
using System.IO;
using System.Net.Http;
using System.Threading.Tasks;
using Railgun.Core.Configuration;

namespace Railgun.Apis.RandomCat
{
    public class RandomCat
    {
        private const string BaseUrl = "http://thecatapi.com/api/images/";
        private readonly string _apiKey;

        public RandomCat(MasterConfig config)
            => _apiKey = config.RandomCatApiToken;

        public async Task<Stream> GetRandomCatAsync()
        {
            Stream stream;

            using (var client = new HttpClient())
            {
                stream = await client.GetStreamAsync($"{BaseUrl}get?api_key={_apiKey}&type=png&size=med");
            }

            return stream;
        }
    }
}